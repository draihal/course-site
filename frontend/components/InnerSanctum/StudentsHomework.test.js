import React from 'react'
import renderer from 'react-test-renderer'
import { render } from '@testing-library/react'
import TestStudentsHomework from './StudentsHomework'

const homework = [
  {
    "id": 1,
    "grade": {
      "id": 1,
      "teacher_last_name": "Test teacher_last_name 1",
      "status": "rework",
      "grade": "test grade 1",
    },
    "student_first_name": "Student 1",
    "student_last_name": "",
    "student_homework": "student_homework 1",
    "lesson": 1,
    "student": 1
  },
  {
    "id": 2,
    "grade": {
      "id": 2,
      "teacher_last_name": "Test teacher_last_name 2",
      "status": "done",
      "grade": "test grade 2",
    },
    "student_first_name": "Student 2",
    "student_last_name": "",
    "student_homework": "student_homework 2",
    "lesson": 1,
    "student": 2
  }
];


describe('StudentsHomework', () => {
  it('render correctly StudentsHomework component', () => {
    const tree = renderer.create(<TestStudentsHomework homework={homework} />).toJSON();
    expect(tree).toMatchSnapshot()
  });

  it('render StudentsHomework correctly with given props', () => {
    const { queryByText } = render(<TestStudentsHomework homework={homework} />);
    expect(queryByText(/Student 1/i)).toBeTruthy();
    expect(queryByText(/Student 2/i)).toBeTruthy();
  });

});