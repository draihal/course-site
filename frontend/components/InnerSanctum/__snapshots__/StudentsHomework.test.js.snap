// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StudentsHomework render correctly StudentsHomework component 1`] = `
<div
  className="row"
>
  <div
    className="col-md-12 table-responsive-md"
  >
    <table
      className="table"
    >
      <thead>
        <tr>
          <th
            scope="col"
          >
            Студент
          </th>
          <th
            scope="col"
          >
            Статус проверки
          </th>
          <th
            scope="col"
          >
            Оценка
          </th>
          <th
            scope="col"
          >
            Проверил
          </th>
          <th
            scope="col"
          >
            Домашняя работа
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <th>
            Student 1
             
            
          </th>
          <th>
            rework
          </th>
          <th>
            test grade 1
          </th>
          <th>
            Test teacher_last_name 1
          </th>
          <th>
            <a
              href="/grade?homework=1"
              onClick={[Function]}
              onMouseEnter={[Function]}
            >
              Проверить
            </a>
          </th>
        </tr>
        <tr>
          <th>
            Student 2
             
            
          </th>
          <th>
            done
          </th>
          <th>
            test grade 2
          </th>
          <th>
            Test teacher_last_name 2
          </th>
          <th>
            <a
              href="/grade?homework=2"
              onClick={[Function]}
              onMouseEnter={[Function]}
            >
              Проверить
            </a>
          </th>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
